src/accountsdb/index.zig

```diff
@@ -1058,7 +1060,81 @@ pub const DiskMemoryAllocator = struct {
     }
 };
 
-test "tests disk allocator on hashmaps" {
+test "swissmap resize" {
+    var map = SwissMap(Pubkey, AccountRef, pubkey_hash, pubkey_eql).init(std.testing.allocator);
+    defer map.deinit();
+
+    try map.ensureTotalCapacity(100);
+
+    const ref = AccountRef.default();
+    map.putAssumeCapacity(Pubkey.default(), ref);
+
+    // this will resize the map with the key still in there
+    try map.ensureTotalCapacity(200);
+    const get_ref = map.get(Pubkey.default()) orelse return error.MissingAccount;
+    try std.testing.expect(std.meta.eql(get_ref, ref));
+}
+
+test "account index update/remove reference" {
+    const allocator = std.testing.allocator;
+
+    var index = try AccountIndex.init(allocator, allocator, 8);
+    defer index.deinit(true);
+    try index.ensureTotalCapacity(100);
+
+    // pubkey -> a
+    var ref_a = AccountRef.default();
+    index.indexRef(&ref_a);
+
+    var ref_b = AccountRef.default();
+    ref_b.slot = 1;
+    index.indexRef(&ref_b);
+
+    // make sure indexRef works
+    {
+        var ref_head_rw = index.getReference(&ref_a.pubkey).?;
+        const ref_head, var ref_head_lg = ref_head_rw.writeWithLock();
+        ref_head_lg.unlock();
+        _, const ref_max = ref_head.highestRootedSlot(10);
+        try std.testing.expect(ref_max == 1);
+    }
+
+    // update the tail
+    try std.testing.expect(ref_b.location == .Cache);
+    var ref_b2 = ref_b;
+    ref_b2.location = .{ .File = .{
+        .file_id = FileId.fromInt(@intCast(1)),
+        .offset = 10,
+    } };
+    try index.updateReference(&ref_b.pubkey, 1, &ref_b2);
+    {
+        const ref = index.getReferenceSlot(&ref_a.pubkey, 1).?;
+        try std.testing.expect(ref.location == .File);
+    }
+
+    // update the head
+    var ref_a2 = ref_a;
+    ref_a2.location = .{ .File = .{
+        .file_id = FileId.fromInt(@intCast(1)),
```

Comment:
Pointless integer cast:
```suggestion
        .file_id = FileId.fromInt(1),
```

Reply:
eb28c8e